test/unit/services/synchronization/nowait_lock_service_test.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class Synchronization::NowaitLockServiceTest < ActionDispatch::IntegrationTest
+  attr_reader :lock_key
+
+  setup do
+    @lock_key = SecureRandom.uuid
+  end
+
+  test "locks then cannot lock but can lock after unlock" do
+    assert Synchronization::NowaitLockService.call(lock_key, timeout: 10000).result
+    assert_not Synchronization::NowaitLockService.call(lock_key, timeout: 10000).result
+    assert_equal 1, Synchronization::UnsafeUnlockService.call(lock_key).result
+    assert_equal 0, Synchronization::UnsafeUnlockService.call(lock_key).result
+    assert Synchronization::NowaitLockService.call(lock_key, timeout: 100).result
+  end
+
+  test "locks between threads" do
+    Thread.new do
+      assert Synchronization::NowaitLockService.call(lock_key, timeout: 10000).result
+    end.join
+    assert_not Synchronization::NowaitLockService.call(lock_key, timeout: 10000).result
```

Comment:
In other tests, you `sleep 0.01` to wait for the first thread to acquire the lock, why not here?

Reply:
Here we call `#join` so we know that the thread has completed.